node_modules/@actions/core/lib/core.js

```diff
@@ -42,8 +42,17 @@ var ExitCode;
  * @param val the value of the variable
  */
 function exportVariable(name, val) {
-    process.env[name] = val;
-    command_1.issueCommand('set-env', { name }, val);
+    const convertedVal = utils_1.toCommandValue(val);
+    process.env[name] = convertedVal;
+    const filePath = process.env['GITHUB_ENV'] || '';
+    if (filePath) {
```

Comment:
I suppose this validates that `filePath` is not null by default?

Reply:
I think it's doing a little bit of error checking to make sure the coding is running in the context of a runner that is updated to use this method of setting env vars. If it's not then it falls back to the old method of using the `set-env` command.